src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -195,27 +195,145 @@ class WorkflowManagerActor(dataAccess: DataAccess, backend: Backend) extends Act
+    // not quite the format required for the metadata REST endpoint.
+    case class AssembledCallMetadata(key: ExecutionDatabaseKey,
+                                     execution: Execution,
+                                     inputs: Traversable[Symbol] = Seq.empty,
+                                     outputs: Option[Traversable[Symbol]] = None,
+                                     streams: Option[StdoutStderr] = None,
+                                     backend: Option[String] = None,
+                                     jobId: Option[String] = None,
+                                     backendStatus: Option[String] = None)
+
+    val executionKeys = executions map { x => ExecutionDatabaseKey(x.callFqn, x.index.toIndex) }
+
+    // Define a sequence of ExecutionMap transformer functions and fold a zero Map across them.
+    type ExecutionMap = Map[ExecutionDatabaseKey, AssembledCallMetadata]
+    type ExecutionMapTransformer = ExecutionMap => ExecutionMap
+
+    // Build the base execution map from scratch.
+    val baseExecutionMapTransformer: ExecutionMapTransformer = _ => {
+      for {
+        execution <- executions
+        key = ExecutionDatabaseKey(execution.callFqn, execution.index.toIndex)
+        if !execution.callFqn.isScatter && !isCollector(executionKeys)(key)
+      } yield key -> AssembledCallMetadata(key, execution)
+    }.toMap
+
+    // Add inputs doctored to include indexes.
+    val inputsTransformer: ExecutionMapTransformer = executionMap => {
+      // Remove collector entries for the purposes of this endpoint.
+      val inputsNoCollectors = filterCollectorSymbols(callInputs)
+      val inputsByKey = inputsNoCollectors.groupBy(i => ExecutionDatabaseKey(i.scope, i.index.toIndex))
+
+      for {
+        (inputKey, inputs) <- inputsByKey
+        // Inputs always have index None, but execution entries may have real indexes.  Look for executions by
+        // FQN only, then give them copies of the inputs with matching indexes.
+        (executionKey, assembledMetadata) <- executionMap
+        if executionKey.fqn == inputKey.fqn
+        indexedInputs = inputs map { _.copy(index = executionKey.index.fromIndex) }
+      } yield executionKey -> assembledMetadata.copy(inputs = indexedInputs)
+    }
+
+    // Add outputs.
+    val outputsTransformer: ExecutionMapTransformer = executionMap => {
+      // Remove collector entries for the purposes of this endpoint.
+      val outputsNoCollectors = filterCollectorSymbols(callOutputs)
+      val outputsByKey = outputsNoCollectors.groupBy(o => ExecutionDatabaseKey(o.scope, o.index.toIndex))
+      for {
+        (key, outputs) <- outputsByKey
+        baseMetadata = executionMap.get(key).get
+      } yield key -> baseMetadata.copy(outputs = Option(outputs))
+    }
+
+    // Add backend-specific job data.
+    val jobDataTransformer: ExecutionMapTransformer = executionMap => {
```

Comment:
These are function values defined as `val`s so they can be put into the `Seq` below for `foldMap`.